dex/networks/eth/params.go

```diff
@@ -65,6 +70,26 @@ var v0Gases = &Gases{
 	RefundGas: 43000,
 }
 
+// EncodeSwapData packs the contract version and the secret hash into a byte
+// slice for communicating a swap's identity.
+func EncodeSwapData(contractVersion uint32, swapKey [SecretHashSize]byte) []byte {
+	b := make([]byte, SecretHashSize+4)
+	binary.BigEndian.PutUint32(b[:4], contractVersion)
+	copy(b[4:], swapKey[:])
+	return b
+}
+
+// DecodeSwapData unpacks the contract version and secret hash.
+func DecodeSwapData(data []byte) (contractVersion uint32, swapKey [SecretHashSize]byte, err error) {
```

Comment:
How about `DecodeVersionedHash` or something? `DecodeSwapData` seems like it could do the same thing as `ParseInitiateData` to me.

Reply:
That's alright with me, but the "versioned" wording is slightly confusing to me because I see this as a piece of data that identifies a swap by `(contract version, secret hash)` rather than a hash that is versioned.  I guess that's pretty subtle.
I was initially wanting this to be worded generically to reflect that it's global swap identifying data, rather than describing that's in the data, but this is all internal to the eth packages, so I suppose there's no need to be too generic, plus we can always rename if there's more data we find necessary to include in this blob.